### PR TITLE
#684 fixed bug when .open calls manually but state aren't changing

### PR DESCRIPTION
**Pull request by ilajosmanov**
Fixed bug with .open event.
Also fixed case when open/close event calls but the payload of these events is incorrect

### CODE REVIEW

File: src/vue/composition.h.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import {Domain, Event, Store} from 'effector'
 
 type Gate<Props> = {
-  open: Event<Props>;
-  close: Event<Props>;
+  open: Event<Props | void>;
```
Comment (igorkamyshev) [inline src/vue/composition.h.ts:4]: 🤔 I do not get it, why `open` can be called without props?

File: src/vue/createGate.ts [hunk 2]
```diff
@@ -34,13 +34,21 @@ export function useGate<Props>(GateComponent: Gate<Props>, cb?: () => Props) {
   }
 
   onMounted(() => {
-    const raw = unwrapProxy(_)
-    GateComponent.open(deepCopy(raw))
+    if (typeof _ !== "undefined") {
```
Comment (igorkamyshev) [inline src/vue/createGate.ts:37]: Same here

File: src/vue/composition.h.ts (hunk 1 above)
Comment (ilajosmanov) [reply src/vue/composition.h.ts:4]: @igorkamyshev why not? Sometimes I can use gate without passing props

File: src/vue/createGate.ts (hunk 2 above)
Comment (ilajosmanov) [reply src/vue/createGate.ts:37]: answer above

File: src/vue/composition.h.ts (hunk 1 above)
Comment (igorkamyshev) [reply src/vue/composition.h.ts:4]: As I understand particular Gate should always be called with props or without props.

So I suggest to use Props type here and pass Props = void, if you are not going to use props.

Comment (ilajosmanov) [reply src/vue/composition.h.ts]: Thanks! Fixed that

**Review by igorkamyshev — APPROVED** [review]
Looks good to me 👍

**Review by zerobias — APPROVED** [review]
Cool, thanks! 👍

**Pull request merged**